packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx

```diff
@@ -0,0 +1,244 @@
+
+    const elementWithTabStop = this.grid.rows[nextRow].cells[nextCol];
+    elementWithTabStop.setAttribute('tabIndex', '0');
+    elementWithTabStop.focus();
+    event.preventDefault();
+  }
+
+  handleClick(event) {
+    // Is this event for something that is on the grid
+    if (event.target?.parentNode?.parentNode?.parentNode?.id !== this.props.id) {
+      return;
+    }
+    // Remove Tab stop from previous cell in table that has focus and set it to the cell that was clicked.
+    this.grid.rows[this.focussedRow].cells[this.focussedCol].setAttribute('tabIndex', '-1');
+    this.focussedRow = event.target.parentElement.rowIndex;
+    this.focussedCol = event.target.cellIndex;
+    const clickedCell = event.target;
+    clickedCell.setAttribute('tabIndex', '0');
+    clickedCell.focus();
+    event.preventDefault();
+  }
+
+  getNumberOfHeaderRows() {
+    return this.props.columnHeaders?.pinnedColumns?.length > 0 ? 1 : 0;
+  }
+
+  getCellData(cell, cellColumnIndex) {
+    const cellClassName = { className: classNames(cx(this.props.rowHeaderIndex === cellColumnIndex ? 'worklist-data-grid-row-header' : 'worklist-data-grid-cell-data')) };
+    const tabIndex = { tabIndex: '-1' };
+    return this.props.rowHeaderIndex === cellColumnIndex ? (<th key={cellColumnIndex} {...tabIndex} {...cellClassName} role="rowheader">{cell.cellContent}</th>) : (<td key={cellColumnIndex} {...tabIndex} {...cellClassName}>{cell.cellContent}</td>);
+  }
+
+  buildColumnHeaders(colHeader) {
+    if (colHeader?.pinnedColumns?.length > 0) {
+      return (
+        <tr height={this.props.columnHeaderHeight}>
+          {colHeader.pinnedColumns.map(columnData => (this.buildSingleColumnHeading(columnData)))}
+        </tr>
+      );
+    }
+    return undefined;
+  }
+
+  buildSingleColumnHeading(columnData) {
+    const width = columnData.width || this.props.columnHeaderDefaultWidth;
+    const height = columnData.height || this.props.columnHeaderHeight;
+    const columnHeaderClassName = classNames(cx('worklist-data-grid-column-header'));
+    return (
+      /* eslint-disable react/forbid-dom-props */
+      <th key={columnData.id} id={columnData.id} role="columnheader" className={columnHeaderClassName} tabIndex="-1" style={{ width, height }}>{columnData.displayName}</th>
+    );
+  }
+
+  buildSingleRow(row, rowIndex) {
+    const backgroundClassName = classNames(cx({ striped: !!(rowIndex % 2) }));
+    return (
+      <tr key={rowIndex} className={backgroundClassName} height={row.height || this.props.rowDefaultHeight}>
+        {row.cells.map((cell, cellColumnIndex) => (
+          this.getCellData(cell, cellColumnIndex)
+        ))}
+      </tr>
+    );
+  }
+
+  buildRows(rows) {
+    return (
+      rows.map((element, index) => (this.buildSingleRow(element, index)))
+    );
+  }
+
+  render() {
+    const {
+      id,
+      ariaLabelledby,
+      ariaLabel,
+      columnHeaders,
+      rows,
+      columnHeaderDefaultWidth,
+      ...customProps
```

Comment:
If we want to allow the class name to be set, we should make it a first class prop instead of leveraging custom props.  However, do consumers need the ability to set the class name on this component?

Reply:
There is no need at this time. We will remove it.